session_test.go

```diff
@@ -78,7 +78,8 @@ func TestGetSession(t *testing.T) {
 			Config:      &Config{},
 			Description: "no configuration or credentials",
 			ExpectedError: func(err error) bool {
-				return errors.Is(err, ErrNoValidCredentialSources)
+				var credentialError NoValidCredentialSourcesError
+				return errors.As(err, &credentialError)
```

Comment:
Should this be `errors.Is()` to just return the boolean? `errors.As()` populates a variable that we're throwing away immediately

Reply:
You would think that (I know I certainly do!), but using `errors.Is()` is for error _value_ checking, so you get this compiler error when trying to check against the type itself, e.g.  `errors.Is(err, NoValidCredentialSourcesError)`:

`type NoValidCredentialSourcesError is not an expression`

Quick example: https://play.golang.org/p/3UcimRc0wJv

Maybe I'll drop in a quick `func IsNoValidCredentialSourceError(err error) bool` since this could potentially be useful to check downstream.